src/module-loader.ts

```diff
@@ -0,0 +1,114 @@
+import * as path from 'path'
+import * as url from 'url'
+
+import {Config as IConfig} from './interfaces'
+import {Plugin as IPlugin} from './interfaces'
+import * as Config from './config'
+
+const getPackageType = require('get-package-type')
+
+/**
+ * Provides a mechanism to use dynamic import / import() with tsconfig -> module: commonJS as otherwise import() gets
+ * transpiled to require().
+ */
+const _importDynamic = new Function('modulePath', 'return import(modulePath)') // eslint-disable-line no-new-func
+
+export default class ModuleLoader {
```

Comment:
I thought your original PR had tests? Is this worth unit testing?

Reply:
There were no added tests. I'm certainly willing to submit tests as well.

Reply:
At the bare minimum, can you add a new [fixture for a esm](https://github.com/oclif/core/tree/master/test/command/fixtures) and a [main test for it here](https://github.com/oclif/core/blob/master/test/command/main.test.ts#L76).